### DISCUSSION TITLE
How many days until my package is merged?

### DISCUSSION THREAD

**Discussion by mukul1729** [void-linux/void-packages#50212] [Q&A] [answered]
Hello, Its my first contribution to void linux packages and wanted to know how many days will it take it to get my pull request merged?

**Comment by Duncaen** [comment, marked as answer]
There is no fixed timeline, it fully depends on what it is, if changes are required and if there is interest.

**Reply by lukeflo** [reply]
Are there any kinds of guidelines regarding the aspects "what it is" and "if there is interest"?

Just asking because there are some PRs for new packages I'm personally *very interested* in which seem to be alright and have passed all tests, but still have not been merged since weeks or month.

Just to pick two as example: `niri` window manager #52794 and `ripgrep-all` #52263 

Can regular users/scarce contributors interest like mine be expressed in any tangible manner, beside comments or reddit posts etc. which can easily be overseen?

This post is by no means meant to be an insult/offensive or anything similiar. I appreciate the voluntary work by all maintainers very much! Void Linux is the best distro I've tried so far.  I just want to know if there is a way to emphasize on some aspects regarding wanted packages and similar stuff.

**Reply by tranzystorekk** [reply]
Just to reiterate on and clarify very commonly raised points:

> Just asking because there are some PRs for new packages I'm personally _very interested_ in which seem to be alright and have passed all tests, but still have not been merged since weeks or month.

We're extremely wary of newpkg PRs, since a lot of contributors' interest and commitment turns out to be fleeting, and then we end up with practically dead-on-arrival packages.

> Can regular users/scarce contributors interest like mine be expressed in any tangible manner, beside comments or reddit posts etc. which can easily be overseen?

We're focused on core team's interest, Void's packages are not managed democratically.

**Reply by lukeflo** [reply]
Thanks for your reply.

> We're focused on core team's interest, Void's packages are not managed democratically.

That is indeed a very helpful and clear definition. Since the plain "interest" often given as answer to such questions is a very general wording. Maybe worth a mention in the guidelines. Plus I understand that its too much stuff to get a democratic decision for every package.

Then I will steadily update my PR and hope it doesn't get burried too far back in the PR list that it is overlooked forever.

Keep up the great work!
